docs/source/backends/json_example.json

```diff
@@ -0,0 +1,154 @@
+{
+  "attributes": {
+    "basePath": {
+      "datatype": "STRING",
+      "value": "/data/%T/"
+    },
+    "iterationEncoding": {
+      "datatype": "STRING",
+      "value": "groupBased"
+    },
+    "iterationFormat": {
+      "datatype": "STRING",
+      "value": "/data/%T/"
+    },
+    "meshesPath": {
+      "datatype": "STRING",
+      "value": "meshes/"
+    },
+    "openPMD": {
+      "datatype": "STRING",
+      "value": "1.1.0"
+    },
+    "openPMDextension": {
+      "datatype": "UINT",
+      "value": 0
+    }
+  },
+  "platform_byte_widths": {
+    "BOOL": 1,
+    "CHAR": 1,
+    "DOUBLE": 8,
+    "FLOAT": 4,
+    "INT": 4,
+    "LONG": 8,
+    "LONGLONG": 8,
+    "LONG_DOUBLE": 16,
+    "SHORT": 2,
+    "UCHAR": 1,
+    "UINT": 4,
+    "ULONG": 8,
+    "ULONGLONG": 8,
+    "USHORT": 2
+  },
+  "subgroups": {
+    "data": {
+      "subgroups": {
```

Comment:
Yes, there is a risk of collisions which in practice should not be too often the case. Nevertheless, rather adding a *dict of attributes* inside an `attributes` group would probably be best (no `__`). This keeps the collisions minimal and at the same time is very readable.

The same is true for data sets and groups, but imho that's fine.

Reply:
>But that's already the case, no?
yep

I changed it now, so attributes keep their separate namespace but datasets and subgroups are now directly linked into the containing group's object, see [here](https://github.com/franzpoeschel/openPMD-api/blob/json-backend/docs/source/backends/json_example.json).

Reply:
Looks good!
Do you think you need the `"object_type": "dataset"` and `"object_type": "group"` keys?

Reply:
I added those for the `listPaths` and `listDatasets` operations. I did not want to rely on checking the structure of JSON objects to distinguish groups from dataset, so I made the distinction explicit.
I mean, I both layouts (i.e. of groups vs. datasets) should actually be computationally distinguishable, but should we go that length?

Reply:
I get the first part, but think it's probably ok to check if a `data` members is present.
Can you elaborate what you mean with the last sentence?

Reply:
>Can please you elaborate what you mean with the last sentence (computationally distinguishable)?

Pretty much what you said already, datasets and groups can be distinguished by inspecting their members.
>think it's probably ok to check if a data members is present

Probably not a good idea if you look at [line 28](https://github.com/franzpoeschel/openPMD-api/blob/json-backend/docs/source/backends/json_example.json) of the example JSON file ;)
The safest bet would probably be to check whether the `extent` key exists *and* whether it points to a JSON array. That should work, but it *feels* like a hack. Should I do it this way?

Reply:
Why do we need `extent`? Can't we determine this from a list (of lists (of lists)) by querying their lengths, even in ND-cases?

Would checking for "contains a data member that contains a list" be enough to detect a low-level "data set" vs. low-level "group"?

I just want to avoid introducing too many non-standard attributes/sub-groups/naming conventions. It might be that we need some, since we don't have as in XML/HDF5/ADIOS/... attributes on a path that are distinct from attributes on data sets, but we should keep it minimal, e.g. for attributes it's reasonable and as a "data" key.

Reply:
>Why do we need extent? Can't we determine this from a list (of lists (of lists)) by querying their lengths, even in ND-cases?

In principle, yes. It would just require pre-filling the dataset with `null` values to be sure that it does actually have the specified dimensions. (Currently, the data array is just as big as necessary for holding the chunks that have actually been written).
Would you prefer this?

Reply:
> pre-filling the dataset with null values

makes probably sense, this only works in serial anyway, right?

Reply:
> The latest commit eliminates the extent field, next up will be the elimination of object_type.

Can you please update the checked-in example file? :)

Reply:
>this only works in serial anyway, right?

right
`object_type` is no more and the example file and documentation are updated.

Reply:
cool, thanks! looks good :)